angr/analyses/decompiler/utils.py

```diff
@@ -331,6 +338,37 @@ def first_nonlabel_node(seq: "SequenceNode") -> Optional[Union["BaseNode", ailme
     return None
 
 
+def remove_labels(graph: nx.DiGraph):
+    new_graph = nx.DiGraph()
+    nodes_map = {}
+    for node in graph:
+        node_copy = node.copy()
+        node_copy.statements = [stmt for stmt in node_copy.statements if not isinstance(stmt, ailment.Stmt.Label)]
+        nodes_map[node] = node_copy
+
+    new_graph.add_nodes_from(nodes_map.values())
+    for src, dst in graph.edges:
+        new_graph.add_edge(nodes_map[src], nodes_map[dst])
+
+    return new_graph
+
+
+def add_labels(graph: nx.DiGraph):
```

Comment:
Where is this method (and also `remove_labels`) called?

I would appreciate it if you can merge `Clinic._insert_block_labels` with this one, and add an option to add labels in-place (instead of returning a new graph).

Reply:
@ltfish `remove_labels` is called inside `decompiler.py` (see changes). `add_labels` is not called anywhere in this branch. I'll remove it for now and readd in a future PR.